src/main/scala/no/ndla/articleapi/integration/SearchApiClient.scala

```diff
@@ -0,0 +1,106 @@
+ *
+ * See LICENSE
+ */
+
+package no.ndla.articleapi.integration
+
+import java.util.concurrent.Executors
+
+import com.typesafe.scalalogging.LazyLogging
+import no.ndla.articleapi.ArticleApiProperties.SearchHost
+import no.ndla.articleapi.model.domain.{Article, ArticleType}
+import no.ndla.articleapi.model.api.SearchException
+import no.ndla.articleapi.service.ConverterService
+import no.ndla.network.NdlaClient
+import org.json4s.Formats
+import org.json4s.ext.EnumNameSerializer
+import org.json4s.native.Serialization.write
+import scalaj.http.{Http, HttpRequest, HttpResponse}
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutorService, Future}
+import scala.util.{Failure, Success, Try}
+
+trait SearchApiClient {
+  this: NdlaClient with ConverterService =>
+  val searchApiClient: SearchApiClient
+
+  class SearchApiClient(SearchApiBaseUrl: String = s"http://$SearchHost") extends LazyLogging {
+
+    private val InternalEndpoint = s"$SearchApiBaseUrl/intern"
+    private val indexTimeout = 1000 * 30
+
+    def indexArticle(article: Article): Article = {
+      implicit val formats: Formats =
+        org.json4s.DefaultFormats +
+          new EnumNameSerializer(ArticleType)
+
+      implicit val executionContext: ExecutionContextExecutorService =
+        ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor)
+
+      val future = postWithData[Article, Article](s"$InternalEndpoint/article/", article)
+      future.onComplete {
+        case Success(Success(_)) =>
+          logger.info(s"Successfully indexed article with id: '${article.id
+            .getOrElse(-1)}' and revision '${article.revision.getOrElse(-1)}' in search-api")
+        case Failure(e) =>
+          logger.error(s"Failed to indexed article with id: '${article.id
+            .getOrElse(-1)}' and revision '${article.revision.getOrElse(-1)}' in search-api", e)
+        case Success(Failure(e)) =>
+          logger.error(s"Failed to indexed article with id: '${article.id
+            .getOrElse(-1)}' and revision '${article.revision.getOrElse(-1)}' in search-api", e)
+      }
+
+      article
+    }
+
+    private def postWithData[A, B <: AnyRef](endpointUrl: String, data: B, params: (String, String)*)(
+        implicit mf: Manifest[A],
+        format: org.json4s.Formats,
+        executionContext: ExecutionContext): Future[Try[A]] = {
+
+      Future {
+        ndlaClient.fetchWithForwardedAuth[A](
+          Http(endpointUrl)
+            .postData(write(data))
+            .timeout(indexTimeout, indexTimeout)
+            .method("POST")
+            .params(params.toMap)
+            .header("content-type", "application/json")
+        )
+      }
+    }
+
+    def deleteArticle(id: Long): Long = {
+      implicit val executionContext: ExecutionContextExecutorService =
+        ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor)
+      val future = delete[Long, Long](s"$InternalEndpoint/article/$id")
```

Comment:
Returnerer dette endepunktet `Long` fra search-api?

Reply:
Nei. Den returnere responsen fra elasticsearch direkte. RequestSuccess[DeleteResponse]

Reply:
Ser at det kan sjå slik ut fra koden. Dette blir gjerne unødvendig komplekst.

Reply:
Men vil ikke det feile og alltid logge `Failed to delete...` når den ikke klarer å deserialisere responsen til en `Long`?

Reply:
Jo, sikkert. Ignorerer responsen fra search-api.